tests/fourier/test_fourier_utils.py

```diff
@@ -12,12 +12,69 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """
-Unit tests for :mod:`fourier` utils functions.
+Tests for the Fourier module helper functions.
 """
-from pennylane.fourier.utils import format_nvec
+import pytest
+import numpy as np
+import pennylane as qml
+from pennylane.fourier.utils import (
+    format_nvec,
+    get_spectrum,
+    join_spectra,
+)
 
 
-def test_format_nvec():
-    """Test the format_nvec function."""
-    assert format_nvec(1) == str(1)
-    assert format_nvec([1, -1]) == " 1 -1"
+@pytest.mark.parametrize(
+    "nvec, exp",
+    [(1, "1"), (-20, "-20"), ((23,), " 23"), ((-1,), "-1"), ((2, -1, 42), " 2 -1  42")],
+)
+def test_format_nvec(nvec, exp):
+    """Test formatting of a tuple of integers into a nice string."""
+    assert format_nvec(nvec) == exp
+
+
+@pytest.mark.parametrize(
+    "spectrum1, spectrum2, expected",
+    [
+        ({0, 1}, {0, 1}, {0, 1, 2}),
+        ({0, 3}, {0, 5}, {0, 2, 3, 5, 8}),
+        ({0, 1, 2}, {0, 1}, {0, 1, 2, 3}),
+        ({0, 0.5}, {0, 1}, {0, 0.5, 1.0, 1.5}),
+        ({0, 0.5}, {0}, {0, 0.5}),
+        ({0}, {0, 0.5}, {0, 0.5}),
+    ],
+)
+def test_join_spectra(spectrum1, spectrum2, expected, tol):
+    """Test that spectra are joined correctly."""
+    joined = join_spectra(spectrum1, spectrum2)
+    assert joined == expected
+    # assert np.allclose(sorted(joined), expected, atol=tol, rtol=0)
+
+
+@pytest.mark.parametrize(
+    "op, expected",
+    [
+        (qml.RX(0.1, wires=0), [0, 1]),  # generator is a class
+        (qml.RY(0.1, wires=0), [0, 1]),  # generator is a class
+        (qml.RZ(0.1, wires=0), [0, 1]),  # generator is a class
+        (qml.PhaseShift(0.5, wires=0), [0, 1]),  # generator is an array
+        (qml.CRX(0.2, wires=[0, 1]), [0, 0.5, 1]),  # generator is an array
+        (qml.ControlledPhaseShift(0.5, wires=[0, 1]), [0, 1]),  # generator is an array
+    ],
+)
+def test_get_spectrum(op, expected, tol):
```

Comment:
Same here, `tol` is not being used (I don't actually see it defined in the file so how are these tests passing?)

Reply:
I have no clue, to be honest :sweat_smile: But I remember asking myself that question (so I believe this was not defined previously neither), maybe there is a global config file setting an "accepted" `tol`? :thinking: 
Will just remove it :)

Reply:
Ah wait, `tol` is used as `atol=tol` below. However, this should make the tests fail even more :D

Reply:
`tol` is defined in `conf.py` :)

Reply:
Aha! It's in `tests/conftest.py`
![Screenshot from 2021-10-05 10-01-15](https://user-images.githubusercontent.com/20772922/135983936-fec13845-d4bf-4e0e-bba8-6b8e23fe2ccc.png)

Reply:
Ah, sorry, I hadn't refreshed the page and not seen your reply :)